mail/resourceAvailability.go

```diff
@@ -0,0 +1,117 @@
+package mail
+
+import (
+	"fmt"
+	"github.com/PPSKSY-Cluster/backend/db"
+	"github.com/go-co-op/gocron"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+)
+
+var s = gocron.NewScheduler(time.Local)
+
+var check = func() {
+	if err := checkNotifications(); err != nil {
+		log.Println(err.Error())
+	}
+}
+
+func InitSchedule() error {
+
+	every, err := strconv.Atoi(os.Getenv("NOTIFICATION_INTERVAL"))
+	if err != nil {
+		return err
+	}
+
+	if s.Len() == 0 {
+		_, err := s.Every(every).
+			Second().
+			Do(check)
+		if err != nil {
+			return err
+		}
+
+		s.StartAsync()
+
+		fmt.Println("Initialized schedule")
+	}
+
+	return nil
+}
+
+func checkNotifications() error {
+	notifications, err := db.GetAllNotifications()
+	if err != nil {
+		return err
+	}
+
+	for _, n := range notifications {
+		if err := checkAvailability(n); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkAvailability(notification db.ReservationNotification) error {
```

Comment:
Imo this function could use more comments. It's kind of difficult to understand what the idea was.

As far as I understand : 
Check each reservation that starts before now in 3 months (meaning also reservations from the past). In the clusterReservations map accumulate how many nodes are reserved each day.
Here I think we don't need to iterate over dates before now (i.e. now is the 18th of july, a reservation that started 2 weeks ago satisfies the condition and now we iterate over those two weeks + 3 months and I think we only need to go over the 3 months period)

keyList at the same time accumulates all days between the earliest start we found and now in 3 months with duplicates.

In the next loop we iterate over the keyList and for each date, starting with the earliest, we check how many nodes have been reserved overall. If at least 2 are free t is incremented and if t is 3 or more we send a notification and stop. 
What does it mean if t is 3 though ?

Reply:
Yup I agree, my mistake.

Keeping duplicates is a dumb mistake on my side. The keylist is kept to have a way to iterate over the map in the order that the dates are in, otherwise we would have no way of knowing if a cluster is available for 3 successive days.

3 means that more than two nodes are available for 3 successive days.

Need to add more comments to the code before merging haha